src/app/doctor/services/usecases_test.go

```diff
@@ -0,0 +1,209 @@
+	sampleUUIDTheressaSiahaan = uuid.Must(uuid.NewRandom())
+	samplePassword = "thestrongestpassword"
+
+	sampleDomainIkoUwais = doctor.Domain{
+		Name:    "dr. Iko Uwais",
+		NIP:     "2022 089281 32",
+		SIP:     "SIP.8198.31.2022",
+		Address: "North Way 61 CA, Antartica",
+		DOB:     utils.ConvertStringToDate("1997-08-19"),
+		Gender:  types.MALE,
+		Polyclinic: doctor.PolyclinicReference{
+			ID:   1,
+			Name: "General",
+		},
+		User: doctor.UserReference{
+			ID:        sampleUUIDIkoUwais,
+			Email:     "dr.ikouwais@example.com",
+			Password:  samplePassword,
+			Role:      types.DOCTOR,
+			CreatedAt: time.Now(),
+			UpdatedAt: time.Now(),
+		},
+	}
+
+	sampleDomainIkoUwaisWithNoUUID = doctor.Domain{
+		Name:    "dr. Iko Uwais",
+		NIP:     "2022 089281 32",
+		SIP:     "SIP.8198.31.2022",
+		Address: "North Way 61 CA, Antartica",
+		DOB:     utils.ConvertStringToDate("1997-08-19"),
+		Gender:  types.MALE,
+		Polyclinic: doctor.PolyclinicReference{
+			ID:   1,
+			Name: "General",
+		},
+		User: doctor.UserReference{
+			ID:        uuid.Nil,
+			Email:     "dr.ikouwais@example.com",
+			Password:  samplePassword,
+			Role:      types.DOCTOR,
+			CreatedAt: time.Now(),
+			UpdatedAt: time.Now(),
+		},
+	}
+
+	sampleDomainList = []doctor.Domain{
+		sampleDomainIkoUwais,
+		{
+			Name:    "dr. Theressa Siahaan",
+			NIP:     "2022 089284 84",
+			SIP:     "SIP.8908.74.2022",
+			Address: "South Way 61 FR, Arctic",
+			DOB:     utils.ConvertStringToDate("1998-01-18"),
+			Gender:  types.FEMALE,
+			Polyclinic: doctor.PolyclinicReference{
+				ID:   1,
+				Name: "General",
+			},
+			User: doctor.UserReference{
+				ID:        sampleUUIDTheressaSiahaan,
+				Email:     "dr.theressasiahaan@example.com",
+				Password:  samplePassword,
+				Role:      types.DOCTOR,
+				CreatedAt: time.Now(),
+				UpdatedAt: time.Now(),
+			},
+		},
+	}
+
+	os.Exit(m.Run())
+}
+
+func TestGetAllDoctors(t *testing.T) {
+	t.Run("should got all data", func(t *testing.T) {
+		mockRepo.On("SelectAllData").Return(sampleDomainList, nil).Once()
+		result, err := services.GetAllDoctors()
+
+		assert.Nil(t, err)
+		assert.Greater(t, len(result), 0)
+	})
+
+	t.Run("should got server error", func(t *testing.T) {
+		mockRepo.On("SelectAllData").Return(nil, errors.New("can't connect to the database")).Once()
+		result, err := services.GetAllDoctors()
+
+		assert.NotNil(t, err)
+		assert.Nil(t, result)
+	})
+}
+
+func TestGetDoctorByID(t *testing.T) {
+	t.Run("should got data by id", func(t *testing.T) {
+		mockRepo.On("SelectDataByID", sampleUUIDIkoUwais.String()).Return(&sampleDomainIkoUwais, nil).Once()
+		result, err := services.GetDoctorByID(sampleUUIDIkoUwais.String())
+
+		assert.Nil(t, err)
+		assert.Equal(t, sampleDomainIkoUwais.Name, result.Name)
+	})
+
+	t.Run("should got database error", func(t *testing.T) {
+		mockRepo.On("SelectDataByID", sampleUUIDIkoUwais.String()).Return(nil, errors.New("can't connect to the database")).Once()

```

Comment:
prefer to add Return in a new line, so the reviewer doesn't need to swipe to see the whole code